src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationProvider.cs

```diff
@@ -81,16 +79,15 @@ public virtual void Load()
                         kv.Key.StartsWith(parentPath, StringComparison.OrdinalIgnoreCase) &&
                         kv.Key[parentPath.Length] == ':')
                     {
-                        results.Add(Segment(kv.Key, parentPath.Length + 1));
+                        yield return Segment(kv.Key, parentPath.Length + 1);
                     }
                 }
             }
 
-            results.AddRange(earlierKeys);
-
-            results.Sort(ConfigurationKeyComparer.Comparison);
```

Comment:
Since this is a `public` API, aren't we breaking it by changing the returned results to no longer be sorted?

Maybe we should split this logic into a `public` API that keeps the existing behavior, and an `internal` API that doesn't do sorting that we can call from `GetChildrenImplementation` below and do the sorting on the outside.

Reply:
Great idea!

Reply:
Wait a minutes...In `GetChildrenImplementation` we are referencing the interface `IConfigurationProvider` not the abstract class `ConfigurationProvider`.